src/GafferScene/RendererAlgo.cpp

```diff
@@ -980,6 +1013,20 @@ struct LocationOutput
 				updatedAttributes->members()[g_setsAttributeName] = boost::const_pointer_cast<InternedStringVectorData>( setsAttribute );
 			}
 
+			for( auto &i : updatedAttributes->members() )
+			{
+				const ShaderNetwork *shaderAttribute = IECore::runTimeCast<ShaderNetwork>( i.second.get() );
+				if( shaderAttribute )
+				{
+					ConstShaderNetworkPtr substituted = RendererAlgo::substitutedShader(
+						shaderAttribute, updatedAttributes.get()
+					);
+
+					// const_cast is safe because we're about to assign to m_attributes which is const
+					i.second = const_cast<ShaderNetwork*>( substituted.get() );
```

Comment:
Again, I don't think we can overwrite our own attributes here, because our children will inherit them with substitutions baked in prematurely.